applications/NXmx.nxdl.xml

```diff
@@ -142,6 +142,46 @@
                         <dim index="4" value="k" required="false"/>
                     </dimensions>
                 </field>
+
+                <field name="data_scaling_factor" type="NX_FLOAT" optional="true">
+                    <doc>
+                        An optional scaling factor to apply to the values in ``data``.
+
+                        The elements in ``data`` are often stored as integers for efficiency reasons and need
```

Comment:
Maybe a datatype narrower than floats

Reply:
Not sure what you mean here.  data_scaling_factor and data_offset are both NX_FLOAT.  What data type is narrower than float?

Reply:
I think I meant after correction the result could just be integer if scale and offset were also integers

Reply:
Fixed, upgraded to NX_NUMBER to allow ints or floats.  This is the same type as the DATA field anyways.